skimage/transform/_warps.py

```diff
@@ -58,36 +66,25 @@ def resize(image, output_shape, order=1, mode='constant', cval=0, clip=True,
     (100, 100)
 
     """
-
-    rows, cols = output_shape[0], output_shape[1]
-    orig_rows, orig_cols = image.shape[0], image.shape[1]
-
-    row_scale = float(orig_rows) / rows
-    col_scale = float(orig_cols) / cols
-
-    # 3-dimensional interpolation
-    if len(output_shape) == 3 and (image.ndim == 2
-                                   or output_shape[2] != image.shape[2]):
-        dim = output_shape[2]
-        orig_dim = 1 if image.ndim == 2 else image.shape[2]
-        dim_scale = float(orig_dim) / dim
-
-        map_rows, map_cols, map_dims = np.mgrid[:rows, :cols, :dim]
-        map_rows = row_scale * (map_rows + 0.5) - 0.5
-        map_cols = col_scale * (map_cols + 0.5) - 0.5
-        map_dims = dim_scale * (map_dims + 0.5) - 0.5
-
-        coord_map = np.array([map_rows, map_cols, map_dims])
-
-        image = _convert_warp_input(image, preserve_range)
-
-        out = ndimage.map_coordinates(image, coord_map, order=order,
-                                      mode=mode, cval=cval)
-
-        _clip_warp_output(image, out, order, mode, cval, clip)
-
-    else:  # 2-dimensional interpolation
-
+    output_shape = np.asarray(output_shape)
+    orig_shape = image.shape
+    if len(orig_shape) < len(output_shape):
+        orig_shape = orig_shape + (1, ) * (len(output_shape) - len(orig_shape))
+    elif len(output_shape) < len(orig_shape):
+        orig_shape = orig_shape[:-1]
+    orig_shape = np.asarray(orig_shape)
+
+    dim_scales = orig_shape.astype('f8') / output_shape
```

Comment:
Urm, why not `orig_shape = np.array(orig_shape, dtype=float)` above, to avoid this `'f8'` business?

Indeed I have no idea what you're doing in this whole paragraph; you might want to sprinkle a few comments in there. =)